src/models/appmodel.ts

```diff
@@ -11,17 +12,39 @@ export class AppModel implements IAppModel {
 
     @observable private _selectedFlavor = "";
     get selectedFlavor(): string { return this._selectedFlavor; }
-    set selectedFlavor(item: string) { this._selectedFlavor = item; }
+    set selectedFlavor(item: string) { 
+        this._selectedFlavor = item; 
+        if(!this._isConstructing) this.saveState();
+    }
 
+    private _isConstructing = true;q
     @observable private _textInput = "";
     get flavorInput(): string { return this._textInput; }
     set flavorInput(value: string) { this._textInput = value; }
 
     get flavorTextIsValid() { return Boolean(this.flavorInput); }
 
-    constructor()
+    private _dataModel: IDataModel;
+
+    constructor(dataModel: IDataModel)
     {
         this.selectedFlavor = this.flavors[0];
+        this._dataModel = dataModel;
+        const data = dataModel.load();
+        if(data != null && data != "")
+        {
+            this.flavors.clear();
+            const jsonData = JSON.parse(data);
+            this.selectedFlavor = jsonData.selected;
+            jsonData.flavors.forEach((f: any) => {
+                this.flavors.push(f);
+            });
+            if(this.selectedFlavor == null || this.selectedFlavor === 'undefined') {
```

Comment:
if we have the undefined like a Type, not a string, and  we just want to check that we didn't select any flavor we could use `if (!this.selectFlavor) {...}` 
but I couldn't understand why the undefined is a string? is this regarding Combobox specific?
and why we are using non-strict comparison with the Null but the string with the 'undefined' string?

Reply:
cool

The logic here is:  If there is no selected flavor specified in the saved data, then just select the first one.

Reply:
could you, please, tell me in what use-cases the selectedFlavor might be 'undefined' string?
I'll try to handle this somehow and simplify the condition